dll/win32/shell32/wine/shellpath.c

```diff
@@ -70,8 +70,8 @@ DoGetProductType(PNT_PRODUCT_TYPE ProductType)
     static const WCHAR ProductOptions[] = L"SYSTEM\\CurrentControlSet\\Control\\ProductOptions";
     HKEY hKey;
     LONG error;
-    WCHAR szValue[32];
-    DWORD cbValue, dwType;
+    WCHAR szValue[9];
+    DWORD cbValue;
     static DWORD s_dwProductType = 0;
```

Comment:
@katahiromz, while here, the type is not cached on key error, but is cached on value error.
Is that actually wanted? Or no cache, or always cache, or cache on value success only?
And similar question about the returned value.

Reply:
As you like.

Reply:
> As you like.

Ok, I'll update its behavior, probably as a follow-up PR...